cmake/generic.cmake

```diff
@@ -316,5 +316,5 @@ endfunction(go_test)
 # go_extern(target_name extern_source)
 # go_extern(go_redis github.com/hoisie/redis)
 function(go_extern TARGET_NAME)
-  add_custom_target(${TARGET_NAME} env GOPATH=${GOPATH} ${CMAKE_Go_COMPILER} get ${ARGN})
+  add_custom_target(${TARGET_NAME} env GOPATH=${GOPATH} ${CMAKE_Go_COMPILER} get -d ${ARGN})
```

Comment:
@helinwang Also should we use `Godeps` to setup static dependencies, so we can avoid when the external lib updates and deprecates some API we are using.